internal/service/rds/cluster.go

```diff
@@ -541,14 +562,15 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 
 	if _, ok := d.GetOk("snapshot_identifier"); ok {
 		opts := rds.RestoreDBClusterFromSnapshotInput{
-			CopyTagsToSnapshot:   aws.Bool(d.Get("copy_tags_to_snapshot").(bool)),
-			DBClusterIdentifier:  aws.String(identifier),
-			DeletionProtection:   aws.Bool(d.Get("deletion_protection").(bool)),
-			Engine:               aws.String(d.Get("engine").(string)),
-			EngineMode:           aws.String(d.Get("engine_mode").(string)),
-			ScalingConfiguration: ExpandClusterScalingConfiguration(d.Get("scaling_configuration").([]interface{})),
-			SnapshotIdentifier:   aws.String(d.Get("snapshot_identifier").(string)),
-			Tags:                 Tags(tags.IgnoreAWS()),
+			CopyTagsToSnapshot:               aws.Bool(d.Get("copy_tags_to_snapshot").(bool)),
+			DBClusterIdentifier:              aws.String(identifier),
+			DeletionProtection:               aws.Bool(d.Get("deletion_protection").(bool)),
+			Engine:                           aws.String(d.Get("engine").(string)),
+			EngineMode:                       aws.String(d.Get("engine_mode").(string)),
```

Comment:
I'm guessing the availability of `ServerlessV2ScalingConfiguration` is the one that determines if its v1 or v2 thats configured rather than engine mode. But I think this causes issues in that when a new version appears we'll have to add `ServerlessV3ScalingConfiguration` rather than adapting the schema and adding an engine mode. Again I'm not super knowledgable here just high level observations